src/event-cache/__tests__/EventCache.test.ts

```diff
@@ -192,6 +483,41 @@ describe('EventCache tests', () => {
         expect(eventCache.hasEvents()).toBeFalsy();
     });
 
+    test('when candidate cache size reached, recordEvent drops the newest candidate', async () => {
```

Comment:
question: is it standard to drop the newest candidate? i assumed that the oldest is dropped